docs/interact-with-the-stack.md

```diff
@@ -0,0 +1,163 @@
+
+## Stopping services
+
+All services can be stopped with:
+
+```bash
+make stop-all
+```
+
+Individual services can be stopped using the docker command:
+```bash
+docker container stop candigv2_<name of module>_1
+```
+eg. to stop the ingest container this would be:
+```
+docker container stop candigv2_candig-ingest_1
+```
+
+## Starting services
+
+Logging must be started first, postgres should be started before any relying services
+
+When all containers are stopped the following command can be used to start all CanDIGv2 containers
+
+```
+make start-all
+```
+
+To start a single container, the following docker command can be used:
+
+```
+docker container start candigv2_<name of module>_1
+```
+e.g. for the ingest container:
+```
+docker container start candigv2_candig-ingest_1
+```
+
+## Cleaning and rebuilding individual services
+
+If any individual services are updated, they will need to be cleaned, rebuilt and recomposed. 
+
+Any individual services can be cleaned with:
+
+```bash
+make clean-<name of module>
+```
+
+for example:
+
+```bash
+make clean-htsget
+```
+
+This stops the container, deletes the container and deletes the image.
+
+> [!NOTE]
+> For services that use the postgres container to save data, i.e. htsget (genomic data) and katsu (clinical data), deleting and rebuilding the service will not delete the data in postgres. If there have been changes to the underlying database, the postgres database will need to be deleted and rebuilt. 
+
+To rebuild and recompose a service first run:
+
+```bash
+make build-<name of module>
+```
+
+> [!NOTE]
+> Containers that have an associated volume will need to have the volume rebuild with `make docker-volumes` before being able to successfully compose the container.
+
+Then compose the container with:
+
+```bash
+make compose-<name of module>
+```
+
+> [!IMPORTANT]
+> Some services can't be rebuilt individually without causing issues with the stack, if you are facing issues with modules related to auth, it is recommended to rebuild the entire stack to ensure everything is in sync.
+
+## Non-destructive Rebuild
+
+To rebuild the CanDIGv2 without destroying data in postgres or keycloak the make target `rebuild-keep-data` with:
+
+```bash
+make rebuild-keep-data
+```
+
+> [!NOTE]
+> If there are changes that have changed the structure of the database or impacted the versions of other CANDIG_DATA_MODULES this way of rebuilding cannot be used.
+
+## Destructive Cleanup 
+
+Use the following steps to clean up running CanDIGv2 services in a docker-compose configuration. 
+
+> [!CAUTION] 
+> Note that these steps are destructive and will remove **ALL** logs, containers, secrets, volumes, networks, certs, and images. If you are using docker in a shared environment (i.e. with other non-CanDIGv2 containers running) please consider running the cleanup steps manually instead.
```

Comment:
Er -- the only one of these that can affect things outside of CanDIG is the clean-images step; all of the rest of the steps are either cleaning up something in the temp directory (logs, secrets), or specifically target `label=candigv2` (containers),  `dangling=true` (volumes), or refer to the `.env` (compose)

Reply:
This is something that was already in the docs, are you able to suggest a change to it if it isn't correct?